### PR TITLE
feat(DropdownV2): highlight selected item in drop down

### PR DESCRIPTION
**Pull request by Zhao-Yi-Peng**
Selected item is highlighted with Combobox but not with DropdownV2

**Changed**

- selected item is also highlighted in the list box when open

### CODE REVIEW

File: src/components/DropdownV2/DropdownV2.js [hunk 1]
```diff
@@ -186,7 +186,11 @@ export default class DropdownV2 extends React.Component {
                   <ListBox.MenuItem
                     key={itemToString(item)}
                     isActive={selectedItem === item}
-                    isHighlighted={highlightedIndex === index}
+                    isHighlighted={
+                      highlightedIndex === index ||
+                      (selectedItem && selectedItem.id === item.id) ||
+                      false
```
Comment (emyarod) [inline src/components/DropdownV2/DropdownV2.js]: is this `|| false` expression redundant?

Comment (Zhao-Yi-Peng) [reply src/components/DropdownV2/DropdownV2.js]: Thanks, you are completely right! Stripped the line from Combobox without closer and this was what happened. 🤕

Comment (Zhao-Yi-Peng) [reply src/components/DropdownV2/DropdownV2.js]: Hmm it seems that the expression is needed after all.

Comment (emyarod) [reply src/components/DropdownV2/DropdownV2.js]: that's strange, could you explain why it appears to be needed? I'm not sure I follow

I also checked out your feature branch and removed that part of the expression, and it appears to be functioning as expected

Comment (Zhao-Yi-Peng) [reply src/components/DropdownV2/DropdownV2.js]: Yeah I also did the same and it was working fine. However the CircleCI tests seemed to be complaining with isHighlighted={null}

Comment (emyarod) [reply src/components/DropdownV2/DropdownV2.js]: yeah after taking a look at the new test snapshots I guess we have several options here

explicit null check

```js
isHighlighted={
  highlightedIndex === index ||
  (selectedItem !== null && 
    selectedItem.id &&
    selectedItem.id === item.id)
}
```

type coercion
```js
isHighlighted={
  highlightedIndex === index ||
  (!!selectedItem && 
    selectedItem.id &&
    selectedItem.id === item.id)
}
```

or the original default to `false`

cc @asudoh

Comment (asudoh) [reply src/components/DropdownV2/DropdownV2.js]: Thank you for mulling it over @emyarod! I'm leaning toward converting the entire condition to `boolean` IIUC this topic.

**Pull request merged**